### PR TITLE
✨Implement autoexpand for amp-form textarea element

### LINKED ISSUE #20224: FR: Auto-resizing textarea in amp-form
Would be great to support multiline text inputs that automatically increase in height as the user types. 

Context: `b/122451407`

/to @cvializ /cc @aghassemi

### PR DESCRIPTION
**Pull request by cvializ**
WIP I will add tests before merge. I'm opening the PR now to get earlier feedback

Resolves #20224

See design doc for edge cases considered, goals and non-goals. https://docs.google.com/document/d/1jme4TcAFvyBRCUNnidSRTIdtjYjd0CbQkM4f2NABsLM

If you don't have access, feel free to request and I will share it.

It currently does not support expanding after updates via bind, but if that is part of a major use-case it can be added after this PR.

Before merge:
- [x] Add tests
- [x] ~Prevent autoexpand when initial text content is larger than the initial textarea size~ https://github.com/ampproject/amphtml/issues/20839

### CODE REVIEW

File: validator/validator-main.protoascii [hunk 1]
```diff
@@ -4624,6 +4624,17 @@ tags: {
   html_format: AMP4EMAIL
   html_format: ACTIONS
   tag_name: "TEXTAREA"
+  attrs: {
+    name: "autoexpand"
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: please alphabetize by attrs.name

File: validator/validator-main.protoascii [hunk 2]
```diff
@@ -4624,6 +4624,17 @@ tags: {
...
+    requires_extension: "amp-form"
+  }
+  attrs: {
+    name: "autoshrink-disabled"
```
Comment (choumx) [inline validator/validator-main.protoascii]: Is there any precedence for negative boolean attributes? If not, can we use `autoshrink` that defaults to true instead and require `autoshrink=false` to override?

File: extensions/amp-form/0.1/amp-form-textarea.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+   * @param {!Element} form
+   */
+  constructor(form) {
+    /** @private */
+    this.win_ = devAssert(form.ownerDocument.defaultView);
+
+    /** @private */
+    this.unlisteners_ = [];
+
+    this.unlisteners_.push(listen(form, 'input', e => {
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form-textarea.js]: I might've said something like this in chat, but maybe we can reduce the number of event listeners from O(# of forms) to O(1) by:

1. Only installing a listener if there exists a `textarea[autoexpand]` element in the document (and re-query on DOM_UPDATE).
2. Add `input` event listener on the root element instead of each form (similar to action-impl.js).

File: extensions/amp-form/0.1/amp-form-textarea.js [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+export function installAmpFormTextarea(form) {
+  new AmpFormWithTextarea(form);
+}
+
+/**
+ * Attempt to resize all textarea elements within the given form.
+ * @param {!Element} form
+ */
+function resizeFormTextareaElements(form) {
+  iterateCursor(form.getElementsByTagName('textarea'), element => {
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form-textarea.js]: `HTMLFormElement.elements` to get non-descendant inputs.

File: extensions/amp-form/0.1/amp-form-textarea.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+          !element.hasAttribute(AMP_FORM_TEXTAREA_EXPAND_ATTR)) {
+        return;
+      }
+
+      maybeRemoveResizeBehavior(element);
+    }));
+
+    const throttledResize = throttle(this.win_, () => {
+      resizeFormTextareaElements(form);
+    }, MIN_EVENT_INTERVAL_MS);
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form-textarea.js]: We have `viewport-impl.js#onResize()`.

File: validator/validator-main.protoascii (hunk 2 above)
Comment (cvializ) [reply validator/validator-main.protoascii]: I feel like `blah=false` attribute values are ambiguous and not in line with standard DOM attributes, since you have attributes like `disabled where `<input disabled>`, `<input disabled="disabled">` and `<input disabled="false">` all disable the input, including `disabled=false`. Same with `checked` and `selected`.

I think the API will be cleaner if I remove the `autoshrink-disabled` and re-add it with a different name if there is a demand for it. The vast majority of use-cases should use autoshrink.

File: extensions/amp-form/0.1/amp-form-textarea.js (hunk 4 above)
Comment (cvializ) [reply extensions/amp-form/0.1/amp-form-textarea.js]: If we aren't handling it on an `O(#forms)` basis, then I think I'll use `rootNode.querySelectorAll` to query for the elements
